### ISSUE TITLE
Invalid URI when generates schema from Generics

### ISSUE THREAD

**Issue by sroucheray** [vega/ts-json-schema-generator#273]
Hi and thanks for your good work !

Generating schema from generics produces invalid URIs according to [rfc3986](https://tools.ietf.org/html/rfc3986) because it contains angle brackets which should be escaped.

Example in this [valid test](https://github.com/vega/ts-json-schema-generator/blob/master/test/valid-data/generic-simple/main.ts)  

It generates this
```json
...
"properties": {
    "value": {
        "$ref": "#/definitions/MyGeneric<number>"
    }
},
...
```  
While it would be better to generate this
```json
...
"properties": {
    "value": {
        "$ref": "#/definitions/MyGeneric%3Cnumber%3E"
    }
},
...
```  
It causes issues if the generated schema is to be later validated by other tools, which is my case.

**Comment by thejuan** [comment]
Fixed by #277

**Comment by sroucheray** [comment]
Thanks @thejuan !
